test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-search-block-spec.js

```diff
@@ -0,0 +1,188 @@
+			await driver.findElement( textLocator ).click();
+			const labelToggleButton = By.css( 'button[aria-label="Use button with icon"]' );
+
+			await driverHelper.waitUntilElementLocatedAndVisible( driver, labelToggleButton );
+			await driver.findElement( labelToggleButton ).click();
+
+			return driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__button.has-icon' )
+			);
+		} );
+
+		it( 'Can remove search button', async function () {
+			//html/body/div[1]/div[2]/div[1]/div[1]/div[3]/div[1]/div[2]/div/div/div/div/div/div/button[3]
+			const textLocator = By.css( 'button[aria-label="Change button position"]' );
+			await driverHelper.waitUntilElementLocatedAndVisible( driver, textLocator );
+			await driver.findElement( textLocator ).click();
+			await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__button-position-menu' )
+			);
+			const buttonEl = await driver.findElement( By.xpath( `//span[text()='No Button']` ) );
+			await buttonEl.click();
+
+			return driverHelper.waitUntilElementNotLocated(
+				driver,
+				By.css( '.wp-block-search__button' )
+			);
+		} );
+
+		it( 'Can Search block settings', async function () {
+			const textLocator = By.css( '.wp-block-search__input' );
+			await driverHelper.waitUntilElementLocatedAndVisible( driver, textLocator );
+			await driver.findElement( textLocator ).click();
+
+			const settingsButton = By.css( 'button[aria-label="Settings"]' );
+			await driverHelper.waitUntilElementLocatedAndVisible( driver, settingsButton );
+			await driver.findElement( settingsButton ).click();
+
+			const percentageWidth = By.css( 'div[aria-label="Percentage Width"]' );
+
+			const selectFirst = await driver.findElement( percentageWidth );
+
+			const primarySelect = await selectFirst.findElement( By.xpath( `//button[text()='25']` ) );
+
+			await primarySelect.click();
+
+			return await driverHelper.waitUntilElementLocatedAndVisible(
+				driver,
+				By.css( '.wp-block-search__inside-wrapper' )
+			);
+		} );
+
+		it( 'Can publish and view content', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			// We need to save the post to get a stable post slug for the block's `url` attribute.
+			// See https://github.com/godaddy-wordpress/coblocks/issues/1663.
+			await gEditorComponent.ensureSaved();
+			return await gEditorComponent.publish( { visit: true } );
+		} );
+
+		it( 'Can see the 25% Search block in published post without label and button', async function () {
+			driverHelper.waitUntilElementNotLocated( driver, By.css( '.wp-block-search__label' ) );
+			driverHelper.waitUntilElementNotLocated( driver, By.css( '.wp-block-search__button' ) );
+
+			//Unable to get width of the search block here and neither in edit mode. In edit mode it gets the  min-width:290px
+			//Here in publish mode it get 173px though that style value is not there when I inspect the object
```

Comment:
Just curious what `getRect()`got you since it is part of the Selenium 4 spec. I know these size things can vary greatly so it isn't surprising that it wasn't giving you what you expected.

Reply:
getRect() - This gives height and width  in pixels along with coordinates. It requires mapping 25% to 173, which I felt is not directly understandable. So wanted to use the "width: 25%;" style property directly. So used getAttribute instead as it's self-explanatory.  Here is what I get 

 + {
+   height: 57,
+   width: 173,
+   x: 202.5,
+   y: 347.5
+ }

Reply:
Ok, that makes sense. I was just curious. 😃